src/Streamly/Internal/Data/Parser/ParserD.hs

```diff
@@ -352,16 +352,18 @@ takeEQ n (Fold fstep finitial fextract) = Parser step initial extract
 
     where
 
-    initial = Tuple' 0 <$> finitial
+    initial = Tuple' n <$> finitial
 
-    step (Tuple' i r) a = do
+    step (Tuple' i r) a
+      | i == 0 = Done 1 <$> fextract r
```

Comment:
The comment I made in `take` applies here as well. And the fix is also the same. In the old code just replace `res` with `s1`.